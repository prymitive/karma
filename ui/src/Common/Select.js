const NewLabelName = (v) => `New label: ${v}`;

const NewLabelValue = (v) => `New value: ${v}`;

export { NewLabelName, NewLabelValue };
