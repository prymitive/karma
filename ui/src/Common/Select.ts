export const NewLabelName = (v: string): string => `New label: ${v}`;

export const NewLabelValue = (v: string): string => `New value: ${v}`;

export interface OptionT {
  label: string;
  value: string;
}

export interface MultiValueOptionT {
  label: string;
  value: string[];
}

export const StringToOption = (value: string): OptionT => ({
  label: value,
  value: value,
});
