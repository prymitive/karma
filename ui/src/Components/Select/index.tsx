import { FC } from "react";

import { components, MenuProps } from "react-select";

import { DropdownSlide } from "Components/Animations/DropdownSlide";
import { OptionT, MultiValueOptionT } from "Common/Select";

export const AnimatedMenu: FC<MenuProps<OptionT, false>> = (props) => {
  return (
    <DropdownSlide in unmountOnExit>
      <components.Menu {...props}>{props.children}</components.Menu>
    </DropdownSlide>
  );
};

export const AnimatedMultiMenu: FC<MenuProps<MultiValueOptionT, true>> = (
  props
) => {
  return (
    <DropdownSlide in unmountOnExit>
      <components.Menu {...props}>{props.children}</components.Menu>
    </DropdownSlide>
  );
};
