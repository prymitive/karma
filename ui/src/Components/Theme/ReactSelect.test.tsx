import { mount } from "enzyme";

import toDiffableHtml from "diffable-html";

import Select from "react-select";

import {
  ReactSelectColors,
  ReactSelectStyles,
} from "Components/Theme/ReactSelect";

const Option = (value: string) => ({ label: value, value: value });

const ThemedSelect = (props: any) => (
  <Select styles={ReactSelectStyles(ReactSelectColors.Light)} {...props} />
);

describe("<WrappedCustomMultiSelect />", () => {
  it("matches snapshot with isMulti=true", () => {
    const tree = mount(<ThemedSelect isMulti />);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot when focused", () => {
    // this test is to cover styles state.isFocused conditions
    const tree = mount(<ThemedSelect autoFocus />);
    tree.find("input").simulate("focus");
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot when focused and disabled", () => {
    const tree = mount(<ThemedSelect autoFocus isDisabled />);
    tree.find("input").simulate("focus");
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot with a value", () => {
    const tree = mount(
      <ThemedSelect
        defaultValue={Option("foo")}
        options={[Option("foo"), Option("bar")]}
      />
    );
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot with isMulti=true and a value", () => {
    const tree = mount(
      <ThemedSelect
        isMulti
        defaultValue={Option("foo")}
        options={[Option("foo"), Option("bar")]}
      />
    );
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot with isDisabled=true", () => {
    const tree = mount(
      <ThemedSelect
        isDisabled
        defaultValue={Option("foo")}
        options={[Option("foo"), Option("bar")]}
      />
    );
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });
});
