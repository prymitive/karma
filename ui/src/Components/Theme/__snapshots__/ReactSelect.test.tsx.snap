// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WrappedCustomMultiSelect /> matches snapshot when focused 1`] = `
"
<div class=\\" css-2b097c-container\\">
  <span aria-live=\\"polite\\"
        aria-atomic=\\"false\\"
        aria-relevant=\\"additions text\\"
        class=\\"css-1f43avz-a11yText-A11yText\\"
  >
    <span id=\\"aria-selection\\">
    </span>
    <span id=\\"aria-context\\">
      Select is focused ,type to refine list, press Down to open the menu,
    </span>
  </span>
  <div class=\\" css-11rrdhm-control\\">
    <div class=\\" css-17yk2dy-ValueContainer\\">
      <div class=\\" css-1wa3eu0-placeholder\\">
        Select...
      </div>
      <div class=\\"css-ps6ina-Input\\">
        <div class
             style=\\"display: inline-block;\\"
        >
          <input autocapitalize=\\"none\\"
                 autocomplete=\\"off\\"
                 autocorrect=\\"off\\"
                 id=\\"react-select-3-input\\"
                 spellcheck=\\"false\\"
                 tabindex=\\"0\\"
                 type=\\"text\\"
                 aria-autocomplete=\\"list\\"
                 style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\"
                 value
          >
          <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
          </div>
        </div>
      </div>
    </div>
    <div class=\\" css-vcwr3k-IndicatorsContainer\\">
      <span class=\\" css-1okebmr-indicatorSeparator\\">
      </span>
      <div class=\\" css-1ij327q-indicatorContainer\\"
           aria-hidden=\\"true\\"
      >
        <svg height=\\"20\\"
             width=\\"20\\"
             viewbox=\\"0 0 20 20\\"
             aria-hidden=\\"true\\"
             focusable=\\"false\\"
             class=\\"css-tj5bde-Svg\\"
        >
          <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`<WrappedCustomMultiSelect /> matches snapshot when focused and disabled 1`] = `
"
<div class=\\" css-14jk2my-container\\">
  <span aria-live=\\"polite\\"
        aria-atomic=\\"false\\"
        aria-relevant=\\"additions text\\"
        class=\\"css-1f43avz-a11yText-A11yText\\"
  >
    <span id=\\"aria-selection\\">
    </span>
    <span id=\\"aria-context\\">
      Select is focused ,type to refine list, press Down to open the menu,
    </span>
  </span>
  <div class=\\" css-1tnfzgk-control\\">
    <div class=\\" css-17yk2dy-ValueContainer\\">
      <div class=\\" css-1wa3eu0-placeholder\\">
        Select...
      </div>
      <div class=\\"css-1h1cjkp-Input\\">
        <div class
             style=\\"display: inline-block;\\"
        >
          <input disabled
                 autocapitalize=\\"none\\"
                 autocomplete=\\"off\\"
                 autocorrect=\\"off\\"
                 id=\\"react-select-4-input\\"
                 spellcheck=\\"false\\"
                 tabindex=\\"0\\"
                 type=\\"text\\"
                 aria-autocomplete=\\"list\\"
                 style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\"
                 value
          >
          <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
          </div>
        </div>
      </div>
    </div>
    <div class=\\" css-15bdmde-IndicatorsContainer\\">
      <span class=\\" css-109onse-indicatorSeparator\\">
      </span>
      <div class=\\" css-1ij327q-indicatorContainer\\"
           aria-hidden=\\"true\\"
      >
        <svg height=\\"20\\"
             width=\\"20\\"
             viewbox=\\"0 0 20 20\\"
             aria-hidden=\\"true\\"
             focusable=\\"false\\"
             class=\\"css-tj5bde-Svg\\"
        >
          <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`<WrappedCustomMultiSelect /> matches snapshot with a value 1`] = `
"
<div class=\\" css-2b097c-container\\">
  <span aria-live=\\"polite\\"
        aria-atomic=\\"false\\"
        aria-relevant=\\"additions text\\"
        class=\\"css-1f43avz-a11yText-A11yText\\"
  >
  </span>
  <div class=\\" css-cat782-control\\">
    <div class=\\" css-17yk2dy-ValueContainer\\">
      <div class=\\" css-1wh03ml-singleValue\\">
        foo
      </div>
      <div class=\\"css-ps6ina-Input\\">
        <div class
             style=\\"display: inline-block;\\"
        >
          <input autocapitalize=\\"none\\"
                 autocomplete=\\"off\\"
                 autocorrect=\\"off\\"
                 id=\\"react-select-5-input\\"
                 spellcheck=\\"false\\"
                 tabindex=\\"0\\"
                 type=\\"text\\"
                 aria-autocomplete=\\"list\\"
                 style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\"
                 value
          >
          <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
          </div>
        </div>
      </div>
    </div>
    <div class=\\" css-vcwr3k-IndicatorsContainer\\">
      <span class=\\" css-1okebmr-indicatorSeparator\\">
      </span>
      <div class=\\" css-tlfecz-indicatorContainer\\"
           aria-hidden=\\"true\\"
      >
        <svg height=\\"20\\"
             width=\\"20\\"
             viewbox=\\"0 0 20 20\\"
             aria-hidden=\\"true\\"
             focusable=\\"false\\"
             class=\\"css-tj5bde-Svg\\"
        >
          <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`<WrappedCustomMultiSelect /> matches snapshot with isDisabled=true 1`] = `
"
<div class=\\" css-14jk2my-container\\">
  <span aria-live=\\"polite\\"
        aria-atomic=\\"false\\"
        aria-relevant=\\"additions text\\"
        class=\\"css-1f43avz-a11yText-A11yText\\"
  >
  </span>
  <div class=\\" css-j5h6r7-control\\">
    <div class=\\" css-17yk2dy-ValueContainer\\">
      <div class=\\" css-1wh03ml-singleValue\\">
        foo
      </div>
      <div class=\\"css-1h1cjkp-Input\\">
        <div class
             style=\\"display: inline-block;\\"
        >
          <input disabled
                 autocapitalize=\\"none\\"
                 autocomplete=\\"off\\"
                 autocorrect=\\"off\\"
                 id=\\"react-select-7-input\\"
                 spellcheck=\\"false\\"
                 tabindex=\\"0\\"
                 type=\\"text\\"
                 aria-autocomplete=\\"list\\"
                 style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\"
                 value
          >
          <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
          </div>
        </div>
      </div>
    </div>
    <div class=\\" css-15bdmde-IndicatorsContainer\\">
      <span class=\\" css-109onse-indicatorSeparator\\">
      </span>
      <div class=\\" css-tlfecz-indicatorContainer\\"
           aria-hidden=\\"true\\"
      >
        <svg height=\\"20\\"
             width=\\"20\\"
             viewbox=\\"0 0 20 20\\"
             aria-hidden=\\"true\\"
             focusable=\\"false\\"
             class=\\"css-tj5bde-Svg\\"
        >
          <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`<WrappedCustomMultiSelect /> matches snapshot with isMulti=true 1`] = `
"
<div class=\\" css-2b097c-container\\">
  <span aria-live=\\"polite\\"
        aria-atomic=\\"false\\"
        aria-relevant=\\"additions text\\"
        class=\\"css-1f43avz-a11yText-A11yText\\"
  >
  </span>
  <div class=\\" css-cat782-control\\">
    <div class=\\" css-115r1xl-ValueContainer\\">
      <div class=\\" css-1wa3eu0-placeholder\\">
        Select...
      </div>
      <div class=\\"css-ps6ina-Input\\">
        <div class
             style=\\"display: inline-block;\\"
        >
          <input autocapitalize=\\"none\\"
                 autocomplete=\\"off\\"
                 autocorrect=\\"off\\"
                 id=\\"react-select-2-input\\"
                 spellcheck=\\"false\\"
                 tabindex=\\"0\\"
                 type=\\"text\\"
                 aria-autocomplete=\\"list\\"
                 style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\"
                 value
          >
          <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
          </div>
        </div>
      </div>
    </div>
    <div class=\\" css-vcwr3k-IndicatorsContainer\\">
      <span class=\\" css-1okebmr-indicatorSeparator\\">
      </span>
      <div class=\\" css-tlfecz-indicatorContainer\\"
           aria-hidden=\\"true\\"
      >
        <svg height=\\"20\\"
             width=\\"20\\"
             viewbox=\\"0 0 20 20\\"
             aria-hidden=\\"true\\"
             focusable=\\"false\\"
             class=\\"css-tj5bde-Svg\\"
        >
          <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`<WrappedCustomMultiSelect /> matches snapshot with isMulti=true and a value 1`] = `
"
<div class=\\" css-2b097c-container\\">
  <span aria-live=\\"polite\\"
        aria-atomic=\\"false\\"
        aria-relevant=\\"additions text\\"
        class=\\"css-1f43avz-a11yText-A11yText\\"
  >
  </span>
  <div class=\\" css-cat782-control\\">
    <div class=\\" css-115r1xl-ValueContainer\\">
      <div class=\\"css-owt1hd-multiValue\\">
        <div class=\\"css-xbn6jz\\">
          foo
        </div>
        <div class=\\"css-1ucqin4\\">
          <svg height=\\"14\\"
               width=\\"14\\"
               viewbox=\\"0 0 20 20\\"
               aria-hidden=\\"true\\"
               focusable=\\"false\\"
               class=\\"css-tj5bde-Svg\\"
          >
            <path d=\\"M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z\\">
            </path>
          </svg>
        </div>
      </div>
      <div class=\\"css-ps6ina-Input\\">
        <div class
             style=\\"display: inline-block;\\"
        >
          <input autocapitalize=\\"none\\"
                 autocomplete=\\"off\\"
                 autocorrect=\\"off\\"
                 id=\\"react-select-6-input\\"
                 spellcheck=\\"false\\"
                 tabindex=\\"0\\"
                 type=\\"text\\"
                 aria-autocomplete=\\"list\\"
                 style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\"
                 value
          >
          <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
          </div>
        </div>
      </div>
    </div>
    <div class=\\" css-vcwr3k-IndicatorsContainer\\">
      <div class=\\" css-tlfecz-indicatorContainer\\"
           aria-hidden=\\"true\\"
      >
        <svg height=\\"20\\"
             width=\\"20\\"
             viewbox=\\"0 0 20 20\\"
             aria-hidden=\\"true\\"
             focusable=\\"false\\"
             class=\\"css-tj5bde-Svg\\"
        >
          <path d=\\"M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z\\">
          </path>
        </svg>
      </div>
      <span class=\\" css-1okebmr-indicatorSeparator\\">
      </span>
      <div class=\\" css-tlfecz-indicatorContainer\\"
           aria-hidden=\\"true\\"
      >
        <svg height=\\"20\\"
             width=\\"20\\"
             viewbox=\\"0 0 20 20\\"
             aria-hidden=\\"true\\"
             focusable=\\"false\\"
             class=\\"css-tj5bde-Svg\\"
        >
          <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div>
"
`;
